repos/system_upgrade/el7toel8/actors/checkleftoverpackages/actor.py

```diff
@@ -0,0 +1,44 @@
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import get_installed_rpms
+from leapp.models import LeftoverPackages, TransactionCompleted, InstalledUnsignedRPM, RPM
+from leapp.tags import RPMUpgradePhaseTag, IPUWorkflowTag
+
+
+class CheckLeftoverPackages(Actor):
+    """
+    Check if there are any RHEL 7 packages present after upgrade.
+
+    Actor produces message containing these packages. Message is empty if there are no el7 package left.
+    """
+
+    name = 'check_leftover_packages'
+    consumes = (TransactionCompleted, InstalledUnsignedRPM)
+    produces = (LeftoverPackages,)
+    tags = (RPMUpgradePhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        installed_rpms = get_installed_rpms()
+        if not installed_rpms:
+            return
+
+        to_remove = LeftoverPackages()
+        unsigned = [pkg.name for pkg in next(self.consume(InstalledUnsignedRPM), InstalledUnsignedRPM()).items]
+
+        for rpm in installed_rpms:
+            rpm = rpm.strip()
+            if not rpm:
+                continue
+            name, version, release, epoch, packager, arch, pgpsig = rpm.split('|')
+
+            if 'el7' in release and name not in unsigned:
```

Comment:
This check causes false positives. During the upgrade `leapp-repository-deps-el8` package got installed with complete NEVRA:
```leapp-repository-deps-el8-5.0.0-0.201904110905Z.c9e0b2b.remove_el7_post_upgrade.PR110.el8.noarch```
This triggered the check, as the `release` part of the query was `0.201904110905Z.c9e0b2b.remove_el7_post_upgrade.PR110.el8`.

Reply:
I know about this, it's because of branch name and shouldn't happen anyways but I'm adding check to make sure we don't remove leapp mid-upgrade